### PR TITLE
[release-4.3] Bug 1796822: Support deployment on OpenStack with self-signed certificate

### PR DESCRIPTION
**Pull request by mandre**
This is a backport of #473 including the fix from #480.

### CODE REVIEW

File: install/0000_30_machine-api-operator_09_rbac.yaml [hunk 1]
```diff
@@ -348,3 +363,21 @@ rules:
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: cloud-provider-config-reader
+  namespace: openshift-config
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - configmap
```
Comment (iamemilio) [inline install/0000_30_machine-api-operator_09_rbac.yaml]: /s/configmap/configmaps

Comment (iamemilio) [reply install/0000_30_machine-api-operator_09_rbac.yaml]: https://github.com/openshift/machine-api-operator/pull/480/files

**Comment by enxebre** [comment]
this just need patch manager approval now

**Pull request merged**